cmd/generate/config/rules/vault.go

```diff
@@ -0,0 +1,38 @@
+package rules
+
+import (
+	"github.com/zricethezav/gitleaks/v8/cmd/generate/secrets"
+	"github.com/zricethezav/gitleaks/v8/config"
+)
+
+func VaultServiceToken() *config.Rule {
+	// define rule
+	r := config.Rule{
+		Description: "Vault Service Token",
+		RuleID:      "vault-service-token",
+		Regex:       generateUniqueTokenRegex(`hvs\.[a-z0-9_-]{90,}`),
```

Comment:
@maltemorgenstern this looks good although I do have a reservation about the unbounded-ness of this. Is `90` a known length? If not could this be limited to, say `{90,100}`?

Reply:
Hey @zricethezav,
this regex is based on these links/findings:
- [Changelog](https://github.com/hashicorp/vault/releases/tag/v1.10.0): `Service tokens have been updated to be longer (a minimum of 95 bytes)`
- [Vault Code](https://github.com/hashicorp/vault/blob/main/command/login_test.go#L19): `const minTokenLengthExternal = 91`
- [Vault Code](https://github.com/hashicorp/vault/blob/main/vault/token_store.go#L72): `TokenLength = 24`
- (There were some examples in the Vault docs with different lengths - but I can't find them 🤦‍♂️)

I also generated some tokens for two different non-enterprise clusters:
```
hvs.CAESIM9-Qp-6v7d3aavgGLIcerj_ZZtt_8VuiH-vx4qMI6vtGh4KHGh2cy5DeWdSNWI3VUZ3dzdiMk5nTlN5QUhjdWg
hvs.CAESIAmM6J-cefSIb0OnSwZWsh3jgBvTTrCKKQnJmqHSNJcgGh4KHGh2cy5obDF5T1lIV3RaMnhVZEFMSWJXdmhiWGM

hvs.CAESICckhUMH3xk3mS5OAePh5rJ_wouqQ2PB9MtxXJBYtLBfGh4KHGh2cy5DZkQ5M2pKWDVMZVNOYXJyRkp0TDFGbm8
hvs.CAESIBNQUaZCbzQzYz-SwFNyA2SrM1nJd1lG_uH8eBW5JNTFGh4KHGh2cy5XN3JZNXhZM2k0Unl5WWRFakloR092bFU
```

For our specific setup restricting this to e.g. `{90,100}` would work (or even `{91}` - but I am not sure if works with Vault Enterprise clusters or different setups 🤷‍♂️

Reply:
@maltemorgenstern let's go with `{90,100}` for now and can always iterate if it is resulting in too many false positives or too little true positives

Reply:
Sounds good - I updated the PR.